{
	"id": "CVE-2018-14647",
	"vulnerabilities": [
		{
			"id": "CVE-2018-14647",
			"description": "Python's elementtree C accelerator failed to initialise Expat's hash salt during initialization. This could make it easy to conduct denial of service attacks against Expat by constructing an XML document that would cause pathological hash collisions in Expat's internal data structures, consuming large amounts CPU and RAM. The vulnerability exists in Python versions 3.7.0, 3.6.0 through 3.6.6, 3.5.0 through 3.5.6, 3.4.0 through 3.4.9, 2.7.0 through 2.7.15.",
			"severity": [
				{
					"type": "cvss_v31",
					"source": "security.alpinelinux.org",
					"cvss_v31": {
						"vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
						"base_score": 7.5,
						"base_severity": "HIGH",
						"temporal_score": 7.5,
						"temporal_severity": "HIGH",
						"environmental_score": 7.5,
						"environmental_severity": "HIGH"
					}
				}
			],
			"references": [
				{
					"source": "security.alpinelinux.org",
					"url": "https://security.alpinelinux.org/vuln/CVE-2018-14647"
				},
				{
					"source": "security.alpinelinux.org",
					"url": "http://www.securityfocus.com/bid/105396"
				},
				{
					"source": "security.alpinelinux.org",
					"url": "http://www.securitytracker.com/id/1041740"
				},
				{
					"source": "security.alpinelinux.org",
					"url": "https://access.redhat.com/errata/RHSA-2019:1260"
				},
				{
					"source": "security.alpinelinux.org",
					"url": "https://access.redhat.com/errata/RHSA-2019:2030"
				},
				{
					"source": "security.alpinelinux.org",
					"url": "https://access.redhat.com/errata/RHSA-2019:3725"
				},
				{
					"source": "security.alpinelinux.org",
					"url": "https://bugs.python.org/issue34623"
				},
				{
					"source": "security.alpinelinux.org",
					"url": "https://www.debian.org/security/2018/dsa-4306"
				},
				{
					"source": "security.alpinelinux.org",
					"url": "https://www.debian.org/security/2018/dsa-4307"
				},
				{
					"source": "security.alpinelinux.org",
					"url": "http://lists.opensuse.org/opensuse-security-announce/2020-01/msg00040.html"
				},
				{
					"source": "security.alpinelinux.org",
					"url": "https://lists.debian.org/debian-lts-announce/2019/06/msg00022.html"
				},
				{
					"source": "security.alpinelinux.org",
					"url": "https://lists.debian.org/debian-lts-announce/2019/06/msg00023.html"
				},
				{
					"source": "security.alpinelinux.org",
					"url": "https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-14647"
				},
				{
					"source": "security.alpinelinux.org",
					"url": "https://lists.apache.org/thread.html/r1b103833cb5bc8466e24ff0ecc5e75b45a705334ab6a444e64e840a0%40%3Cissues.bookkeeper.apache.org%3E"
				},
				{
					"source": "security.alpinelinux.org",
					"url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/RBJCB2HWOJLP3L7CUQHJHNBHLSVOXJE5/"
				},
				{
					"source": "security.alpinelinux.org",
					"url": "https://usn.ubuntu.com/3817-1/"
				},
				{
					"source": "security.alpinelinux.org",
					"url": "https://usn.ubuntu.com/3817-2/"
				}
			],
			"published": "2018-09-25T00:29:00Z",
			"modified": "2024-06-13T07:17:36.584373Z"
		}
	],
	"detection": [
		{
			"ecosystem": "alpine:3.10",
			"vulnerable": true,
			"package": {
				"name": "python2"
			},
			"affected": {
				"type": "apk",
				"range": [
					{
						"lt": "2.7.16-r0"
					}
				],
				"fixed": [
					"2.7.16-r0"
				]
			}
		},
		{
			"ecosystem": "alpine:3.11",
			"vulnerable": true,
			"package": {
				"name": "python2"
			},
			"affected": {
				"type": "apk",
				"range": [
					{
						"lt": "2.7.16-r0"
					}
				],
				"fixed": [
					"2.7.16-r0"
				]
			}
		},
		{
			"ecosystem": "alpine:3.12",
			"vulnerable": true,
			"package": {
				"name": "python2"
			},
			"affected": {
				"type": "apk",
				"range": [
					{
						"lt": "2.7.16-r0"
					}
				],
				"fixed": [
					"2.7.16-r0"
				]
			}
		},
		{
			"ecosystem": "alpine:3.6",
			"vulnerable": true,
			"package": {
				"name": "python3"
			},
			"affected": {
				"type": "apk",
				"range": [
					{
						"lt": "3.6.8-r0"
					}
				],
				"fixed": [
					"3.6.8-r0"
				]
			}
		},
		{
			"ecosystem": "alpine:3.7",
			"vulnerable": true,
			"package": {
				"name": "python2"
			},
			"affected": {
				"type": "apk",
				"range": [
					{
						"lt": "2.7.15-r2"
					}
				],
				"fixed": [
					"2.7.15-r2"
				]
			}
		},
		{
			"ecosystem": "alpine:3.7",
			"vulnerable": true,
			"package": {
				"name": "python3"
			},
			"affected": {
				"type": "apk",
				"range": [
					{
						"lt": "3.6.8-r0"
					}
				],
				"fixed": [
					"3.6.8-r0"
				]
			}
		},
		{
			"ecosystem": "alpine:3.8",
			"vulnerable": true,
			"package": {
				"name": "python2"
			},
			"affected": {
				"type": "apk",
				"range": [
					{
						"lt": "2.7.15-r2"
					}
				],
				"fixed": [
					"2.7.15-r2"
				]
			}
		},
		{
			"ecosystem": "alpine:3.8",
			"vulnerable": true,
			"package": {
				"name": "python3"
			},
			"affected": {
				"type": "apk",
				"range": [
					{
						"lt": "3.6.8-r0"
					}
				],
				"fixed": [
					"3.6.8-r0"
				]
			}
		},
		{
			"ecosystem": "alpine:3.9",
			"vulnerable": true,
			"package": {
				"name": "python2"
			},
			"affected": {
				"type": "apk",
				"range": [
					{
						"lt": "2.7.16-r0"
					}
				],
				"fixed": [
					"2.7.16-r0"
				]
			}
		}
	],
	"data_source": "alpine-osv"
}
